### ISSUE TITLE
Enhancement: [require-await] Disable for inherited methods

### ISSUE THREAD

**Issue by stevenwdv** [typescript-eslint/typescript-eslint#5415]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/require-await

### Description

I propose adding an option to disable the require `await` check for methods that are inherited, since some implementations of the inherited method may not need to return a Promise and using `Promise.resolve` is verbose. This would apply to methods of implemented interfaces and extended classes.

### Fail

```typescript
async function f() {return 42}
```

### Pass

```typescript
interface Base {
  doIt(): Promise<number>;
}

interface SomeSyncImpl {
  async doIt() {
    // We don't need to call async methods for this particular implementation
    return 42;
  }
}

//interface SomeAsyncImpl...
```

**Comment by bradzacher** [comment]
Heads up that your example code is wrong. You've (a) declared an interface with a method body and (b) not implemented the interface.

**Comment by bradzacher** [comment]
Duplicate of https://github.com/typescript-eslint/typescript-eslint/issues/1599
